src/library.js

```diff
@@ -894,10 +894,15 @@ LibraryManager.library = {
     return ret|0;
   },
 
-  llvm_memcpy_i32: 'memcpy',
-  llvm_memcpy_i64: 'memcpy',
-  llvm_memcpy_p0i8_p0i8_i32: 'memcpy',
-  llvm_memcpy_p0i8_p0i8_i64: 'memcpy',
+  _memcpy_js__deps: ['memcpy'],
+  _memcpy_js: function(dst, src, num) {
+    return _memcpy(dst, src, num);
+  },
+
+  llvm_memcpy_i32: '_memcpy_js',
+  llvm_memcpy_i64: '_memcpy_js',
+  llvm_memcpy_p0i8_p0i8_i32: '_memcpy_js',
+  llvm_memcpy_p0i8_p0i8_i64: '_memcpy_js',
 
```

Comment:
IIUC, these are needed to work around the linking problem where memcpy etc. was not linked in, but late in codegen we realize we need it?

If so then this should not be necessary for the LLVM wasm backend, and we can ifdef these lines to make sure they don't exist there (if this were brought in for there, it would be a bug).

But regardless, this seems separate from the topic of the main PR here - how does this relate to not declaring the exports? The linker issue should exist regardless?

Reply:
This is not because of a linking issue, but because of https://github.com/kripken/emscripten/pull/7505/files#r234345567.

Adjusted the code to omit this indirection when `DECLARE_ASM_MODULE_EXPORTS=0` is not being used, so that we do not pay cost of these aliases when not needed.